### ISSUE TITLE
Training streaming zipformer diverges at 700-1000 steps

### ISSUE THREAD

**Issue by spacetronics** [k2-fsa/icefall#1884]
I'm trying to train a streaming zipformer model with 2 custom datasets, first one with 160 hours of training data and second one with 37 hours training data. For the first train i use default parameters, however in 700 steps and so on, the loss increased and grad scale seems to be unstable. i used different base-lr 0.035, 0.025, but the issue is still there. I then lowered the base-lr to 0.015 and the first 1500 steps seems fine but then the same issue came up. 
Here's the graph:

![Image](https://github.com/user-attachments/assets/d552a888-420c-4896-9363-11754b3f4a6c)

![Image](https://github.com/user-attachments/assets/8ce95dd3-9f24-4e36-aa81-ff2be7637dc9)

![Image](https://github.com/user-attachments/assets/a00bc32c-58ca-4c17-bc0f-9328b124b606)

It's also where the `Parameter dominating tot_sumsq` and the `Parameters with most larger-than-usual grads` warning came up everytime. I filtered my cuts to remove long and short utterance (removed shorter than 1 second and longer than 20 second). My max duration is 300 seconds. I used ZipSampler to combine the datasets. Can someone guide me on what's happening here?
